web_monitoring/diffing_server.py

```diff
@@ -124,6 +136,23 @@ def write_error(self, status_code, **kwargs):
             self.set_status(response['code'])
         self.finish(response)
 
+    def check_response_for_error(self, response):
+        #Check if the HTTP requests were successfull and handle exeptions
+        if response.error is not None:
+            try:
+                response.rethrow()
+            except (ValueError, IOError):
```

Comment:
This works, but [`IOError` got merged into `OSError` in Python 3.3](https://docs.python.org/3/library/exceptions.html#IOError), so it might be better to use that instead.